src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -314,8 +314,26 @@ def csv(self, path: str) -> DataFrame:
         Returns:
             a :class:`DataFrame` that is set up to load data from the specified CSV file(s) in a Snowflake stage.
         """
+        # infer schema is set to false by default
+        if "INFER_SCHEMA" not in self._cur_options:
+            self._cur_options["INFER_SCHEMA"] = False
+        self._infer_schema = self._cur_options["INFER_SCHEMA"]
+        schema_to_cast, transformations = None, None
+
         if not self._user_schema:
-            raise SnowparkClientExceptionMessages.DF_MUST_PROVIDE_SCHEMA_FOR_READING_FILE()
+            if not self._infer_schema:
+                raise SnowparkClientExceptionMessages.DF_MUST_PROVIDE_SCHEMA_FOR_READING_FILE()
+
+            (
+                schema,
+                schema_to_cast,
+                transformations,
+            ) = self._infer_schema_for_file_format(path, "CSV")
+            if not schema:
+                # if infer schema query fails, use $1, VariantType as schema
+                schema = [Attribute('"$1"', VariantType())]
```

Comment:
probably we should get why infer schema failed and propagate the error to the user

Reply:
logging a warning for this.